### PR TITLE
adding clippy to ci

### PR DESCRIPTION
**Pull request by Courtsilius**

### CODE REVIEW

**Comment by Courtsilius** [comment]
bors ping

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -29,4 +28,27 @@ jobs:
...
+          args: -- -D warnings
+
+  ci:
+    name: CI
+    if: ${{ success() }}
+    runs-on: ubuntu-latest
+    needs:
+      - Clippy
+      - Format
+      - Build
```
Comment (matthiasbeyer) [inline .github/workflows/main.yml:52]:
```suggestion
      - clippy
      - fmt
      - build_and_test
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -29,4 +28,27 @@ jobs:
...
+    steps:
+      - exit 0
```
Comment (matthiasbeyer) [inline .github/workflows/main.yml:54]:
```suggestion
      - run: exit 0
```

**Comment by Courtsilius** [comment]
Danke @matthiasbeyer :)

**Comment by Courtsilius** [comment]
bors merge

**Comment by Courtsilius** [comment]
bors merge

**Comment by Courtsilius** [comment]
bors merge

**Pull request closed without merging**